README.md

```diff
@@ -337,15 +337,16 @@ Multiple proxies support the gRPC-web protocol.
 	$ docker-compose up -d node-server envoy commonjs-client
 	```
 
-2. You can also try the [gRPC-web Go proxy][].
+2. [Nginx](https://www.nginx.com/) has a grpc-web module ([doc](https://nginx.org/en/docs/http/ngx_http_grpc_module.html), [announcement](https://www.nginx.com/blog/nginx-1-13-10-grpc/)). It seems to work with simple configs, according to [user feedbacks](https://github.com/grpc/grpc-web/discussions/1322)), even with server-streaming support ([user feedback](https://github.com/grpc/grpc-web/issues/1391)).
```

Comment:
Neither of these nginx pages mention gRPC-web - are we sure they are referring to gRPC-web and not gRPC?

Reply:
Hmm, you're right.. i think that's a very good point.. Somehow i thought they were — I probably misread initially. Maybe they are not.. 

In any case, there are 2 users reporting grpc-web is working, including the OP and this [confirmation](https://github.com/grpc/grpc-web/discussions/1322#discussioncomment-8318790), and hence i changed the doc.

Maybe i should wait for one extra confirmation on the binary streaming case, but the OP sounded pretty sure in https://github.com/grpc/grpc-web/issues/1391 so i trusted his judgements :)

Reply:
I'm pretty suspicious these threads are not consistently distinguishing gRPC traffic from gRPC-web. For example [`application/grpc`](https://github.com/grpc/grpc-web/discussions/1322#discussioncomment-8318790) is [clearly not gRPC-web](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2). Furthermore, it seems unlikely that Nginx has _somehow_ managed to enable streaming support for the gRPC-web binary format when [your understanding was the binary format is incapable of supporting server-side streaming](https://github.com/grpc/grpc-web/issues/1396#issuecomment-1920218318).

More to the point, it's _impossible_ for Nginx to have enabled streaming support for the gRPC-web binary protocol without collaborating with the gRPC-web project, because the protocol spec directly says this is not supported. So if they have implemented some sort of streaming support it's not to spec.

But my bet (especially given the lack of any mention of gRPC-web in their documentation) is Nginx supports _gRPC_, and people are conflating it with gRPC-web because it's a "web" server that's able to proxy gRPC traffic.

Reply:
> I'm pretty suspicious these threads are not consistently distinguishing gRPC traffic from gRPC-web. For example [`application/grpc`](https://github.com/grpc/grpc-web/discussions/1322#discussioncomment-8318790) is [clearly not gRPC-web](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2).

Yeah i can see your suspicion. I don't deny them.

> Furthermore, it seems unlikely that Nginx has _somehow_ managed to enable streaming support for the gRPC-web binary format when [your understanding was the binary format is incapable of supporting server-side streaming](https://github.com/grpc/grpc-web/issues/1396#issuecomment-1920218318).

This is a very good point.. thanks for pointing it out :)

> More to the point, it's _impossible_ for Nginx to have enabled streaming support for the gRPC-web binary format without collaborating with the gRPC-web project, because the protocol spec directly says this is not supported. So if they have implemented some sort of streaming support it's not to spec.

Yes another very good point.

> But my bet (especially given the lack of any mention of gRPC-web in their documentation) is Nginx supports _gRPC_, and people are conflating it with gRPC-web because it's a "web" server that's able to proxy gRPC traffic.

Yes it's definitely a possibility. Let me loop in the original commenters.

---

@jmzwcn @zhumazhenis Hi! Thanks for previously participating in the [discussion](https://github.com/grpc/grpc-web/discussions/1322) on nginx support.

There's a question now on the validity of the conclusions as above.

Could you help responding to it if there are some misunderstandings?

Thanks in advance!

Reply:
@sampajano any objection to reverting these notes? At best they're unclear and IMO it would be better to remove them and hash out exactly what is known before re-adding them.

Reply:
OK! Since the original commenters are not active at this point, i guess we can revert until upon further discussions :)

Do you want to create a PR to revert this one? Or if you prefer i can do it too.

Thanks! :)

Reply:
Thanks, just cut #1408